### ISSUE TITLE
Make the C++ Parser a C/C++ Parser

### ISSUE THREAD

**Issue by mindless2112** [CJCombrink/SpellChecker-Plugin#39]
It looks like you could just change CPP_SOURCE_FILES_REGEXP_PATTERN. It would be nicer than having to rename my `.c` files to `.cpp` to get spellchecking.

Also, there are other commonly used C++ file extensions -- check out the drop-down list in Qt Creator's Open File dialog. Maybe there's a way to query the language of a document from Qt Creator rather than matching on file extension?

**Comment by mindless2112** [comment]
It looks like you could just change CPP_SOURCE_FILES_REGEXP_PATTERN.  While you're at it, there are other commonly used C++ file extensions -- check out the drop-down list in Qt Creator's Open File dialog.

**Comment by CJCombrink** [comment]
Hi @mindless2112 ,

> change CPP_SOURCE_FILES_REGEXP_PATTERN

Yes this should be something that can be done easily. I did oversee c files and also assume C and C++ parsing should be the same. This week I also encountered that same issue and will update accordingly.

> Maybe there's a way to query the language of a document from Qt Creator rather than matching on file extension?

I will definitely look into this, it might be an elegant solution to replace the RegExp.

I will try to spend some time over the weekend to fix this. I must in any case spend the time to update to QtC 3.4.
